packages/fast-tooling-react/src/css-editor/box-shadow/box-shadow.tsx

```diff
@@ -71,112 +82,106 @@ export default class CSSBoxShadow extends Foundation<
-                            className={get(
-                                this.props,
-                                "managedClasses.cssBoxShadow_opacityInput"
-                            )}
-                            type={"number"}
-                            min={0}
-                            max={100}
-                            step={1}
-                            placeholder={"50"}
-                            value={Math.round(this.state.boxShadowOpacity * 100)}
-                            onChange={this.handleBoxShadowOpacityOnChange}
+                            type={"color"}
+                            style={{ opacity: 0, width: "100%" }}
+                            value={this.state.boxShadowColor}
+                            onChange={this.handleBoxShadowColorOnChange}
+                            disabled={this.props.disabled}
+                            onFocus={this.props.reportValidity}
+                            onBlur={this.props.updateValidity}
                         />
                     </div>
+                    <input
+                        className={cssBoxShadow_opacityInput}
+                        type={"number"}
+                        min={0}
+                        max={100}
+                        step={1}
+                        placeholder={"50"}
+                        value={Math.round(this.state.boxShadowOpacity * 100)}
+                        onChange={this.handleBoxShadowOpacityOnChange}
+                        disabled={this.props.disabled}
+                        onFocus={this.props.reportValidity}
+                        onBlur={this.props.updateValidity}
+                        ref={this.props.elementRef as React.Ref<HTMLInputElement>}
+                    />
                 </div>
-                <div
-                    className={get(
-                        this.props,
-                        "managedClasses.cssBoxShadow_controlRegion"
-                    )}
-                >
-                    <label
-                        className={get(this.props, "managedClasses.cssBoxShadow_label")}
-                    >
-                        X
-                    </label>
+                <div className={cssBoxShadow_controlRegion}>
+                    <label className={cssBoxShadow_label}>X</label>
                     <input
-                        className={get(this.props, "managedClasses.cssBoxShadow_xInput")}
+                        className={cssBoxShadow_xInput}
                         type={"text"}
                         placeholder={"5px"}
                         value={this.state.boxShadowOffsetX}
                         onChange={this.handleBoxShadowXOnChange}
+                        disabled={this.props.disabled}
+                        onFocus={this.props.reportValidity}
+                        onBlur={this.props.updateValidity}
+                        ref={this.props.elementRef as React.Ref<HTMLInputElement>}
```

Comment:
I am confused by this, you assign the elementRef to a bunch of different inputs, it should only be on a single input. You also report validity on all of the inputs. This is a 1 to 1 model of validation to input. You should probably address this by breaking up the data so going in it uses an object with keys corresponding to each of these inputs.